### ISSUE TITLE
Web components with ShadowDOM, unable to focus on elements in slot

### ISSUE THREAD

**Issue by sandranrivas** [vueuse/vueuse#3279]
### Describe the bug

I'm using usefocustrap in my web components but I am unable to make the component work with elements within a ShadowDOM slot. Is there a way to have a single focus trap contain elements from both the ShadowDOM and elements passed in via the slot? 

**basicDialog.vue**
```
        <div ref="target">
                <!-- @slot Use this slot for primary action -->
                <slot name="primary-action"></slot>
           </div>
```
```
    const { hasFocus, activate, deactivate } = useFocusTrap(target, {
      immediate: true,
    })
```

**App.vue**
```
      <basic-dialog v-if="show">
        <!-- this is my custom button, which already have a tabindex=0-->
        <my-custom-button-vue-web-component slot="primary-action">
          <div @click="show = false">Cancel</div>
        </my-custom-button-vue-web-component>
      </basic-dialog>
```

### Reproduction

.

### System Info

```Shell
.
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by coderwei99** [comment]
i konw

**Comment by antfu** [comment]
We temporarily close this due to the lack of enough information.
Please provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) to reopen the issue.
Thanks.

> [Why reproduction is required](https://antfu.me/posts/why-reproductions-are-required)

**Comment by sandranrivas** [comment]
it is literally just the title that says everything, there are no support for slots. this is also not documented anywhere
